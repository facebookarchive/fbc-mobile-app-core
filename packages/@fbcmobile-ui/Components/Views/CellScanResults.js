/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import * as React from 'react';
import CellSignalIcon from '@fbcmobile/ui/Components/CellSignalIcon';
import Text from '@fbcmobile/ui/Components/Core/Text';
import {StyleSheet, View} from 'react-native';

export type CellularNetworkType =
  | 'CDMA'
  | 'GSM'
  | 'LTE'
  | 'WCDMA'
  | '%future added value';

export type CellScanData = {|
  networkType: CellularNetworkType,
  signalStrength: number,
  timestamp?: ?number,
  baseStationID?: ?string,
  networkID?: ?string,
  systemID?: ?string,
  cellID?: ?string,
  locationAreaCode?: ?string,
  mobileCountryCode?: ?string,
  mobileNetworkCode?: ?string,
  primaryScramblingCode?: ?string,
  operator?: ?string,
  arfcn?: ?number,
  physicalCellID?: ?string,
  trackingAreaCode?: ?string,
  timingAdvance?: ?number,
  earfcn?: ?number,
  uarfcn?: ?number,
  latitude?: ?number,
  longitude?: ?number,
  altitude?: ?number,
  heading?: ?number,
  rssi?: ?number,
|};

type Props = {|
  +scanResults: Array<CellScanData>,
|};

const CellScanResults = ({scanResults}: Props) => {
  return (
    <View style={styles.root}>
      {scanResults.map((cell, index) => {
        if (cell != null) {
          let cellDescription = cell.networkType;
          if (cell.operator != null) {
            cellDescription = `${cellDescription}, ${cell.operator}`;
          }
          return (
            <View style={styles.entry} key={index}>
              <Text style={styles.cellDescription} color="gray">
                {`${cellDescription}`}
              </Text>
              <CellSignalIcon signalStrength={cell.signalStrength} />
            </View>
          );
        }
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    display: 'flex',
    flexDirection: 'column',
  },
  entry: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingVertical: 12,
  },
  cellIcon: {
    marginRight: 4,
  },
  cellDescription: {
    flexGrow: 1,
  },
});

export default CellScanResults;
