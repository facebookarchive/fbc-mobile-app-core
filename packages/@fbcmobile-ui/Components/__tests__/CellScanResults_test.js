/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import CellScanResults from '../Views/CellScanResults';
import React from 'react';
import renderer from 'react-test-renderer';
import type {CellScanData} from '../Views/CellScanResults';

describe('CellScanResults Tests', () => {
  test('empty CellScanResults renders correctly', () => {
    const tree = renderer.create(<CellScanResults scanResults={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('CellScanResults renders correctly', () => {
    const results: CellScanData = {
      networkType: 'CDMA',
      signalStrength: 1,
      operator: 'myCell',
    };
    const tree = renderer
      .create(<CellScanResults scanResults={[results]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
