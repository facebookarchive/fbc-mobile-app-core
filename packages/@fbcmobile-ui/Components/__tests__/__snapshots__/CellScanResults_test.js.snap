// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellScanResults Tests CellScanResults renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "color": "#484848",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
            },
            Object {
              "color": "#303846",
            },
            null,
            Object {
              "textAlign": "left",
            },
          ],
          Object {
            "flexGrow": 1,
          },
        ]
      }
    >
      CDMA, myCell
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#484848",
            "fontSize": 16,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ó°¢¾
    </Text>
  </View>
</View>
`;

exports[`CellScanResults Tests empty CellScanResults renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
/>
`;
